core/src/streams.rs

```diff
@@ -161,4 +262,364 @@ impl<'gc> NetStream<'gc> {
+                            write.preload_offset = 3;
+                            return;
+                        }
+                    }
+                }
+                Some(magic) => {
+                    //Unrecognized signature
+                    //TODO: Fire an error event to AS & stop playing too
+                    tracing::error!("Unrecognized file signature: {:?}", magic);
+                    write.preload_offset = 3;
+                    return;
+                }
+                None => return, //Data not yet loaded
+            }
+        }
+
+        let end_time = write.stream_time + dt;
+        let mut end_of_video = false;
+        let mut error = false;
+
+        //At this point we should know our stream type.
+        if matches!(write.stream_type, Some(NetStreamType::Flv { .. })) {
+            let mut reader = FlvReader::from_parts(&write.buffer, write.offset);
+
+            loop {
+                let tag = FlvTag::parse(&mut reader);
+                if let Err(e) = tag {
+                    if matches!(e, FlvError::EndOfData) {
+                        //TODO: Check expected total length for streaming / progressive download
+                        end_of_video = true;
+                    } else {
+                        //Corrupt tag or out of data
+                        tracing::error!("FLV tag parsing failed: {}", e);
+                        error = true;
+                    }
+
+                    break;
+                }
+
+                let tag = tag.expect("valid tag");
+                if tag.timestamp as f64 >= end_time {
+                    //All tags processed
+                    if let Err(e) = FlvTag::skip_back(&mut reader) {
+                        tracing::error!("FLV skip back failed: {}", e);
+                    }
+
+                    break;
+                }
+
+                let tag_needs_preloading = reader.stream_position().expect("valid position")
+                    as usize
+                    >= write.preload_offset;
+
+                match tag.data {
+                    FlvTagData::Audio(FlvAudioData { .. }) => {
+                        tracing::warn!("Stub: Stream audio processing");
+                    }
+                    FlvTagData::Video(FlvVideoData { codec_id, data, .. }) => {
+                        let (video_handle, frame_id) = match write.stream_type {
+                            Some(NetStreamType::Flv {
+                                stream, frame_id, ..
+                            }) => (stream, frame_id),
+                            _ => unreachable!(),
+                        };
+                        let codec = VideoCodec::from_u8(codec_id as u8);
+
+                        match (video_handle, codec, data) {
+                            (Some(video_handle), Some(codec), FlvVideoPacket::Data(data)) => {
+                                // NOTE: Currently, no implementation of the decoder backend actually requires
```

Comment:
Not for simple playback, but for discovering which frames are keyframes, and thus, for seeking to work at all.